aiida/restapi/common/utils.py

```diff
@@ -146,63 +146,79 @@ def parse_path(self, path_string):
         :param path_string: the path string
         :return:resource_type (string)
                 page (integer)
-                pk (integer)
+                id (integer or uuid)
                 result_type (string))
         """
+        import uuid
 
         ## Initialization
         page = None
-        pk = None
+        id = None
         query_type = "default"
         path = self.split_path(self.strip_prefix(path_string))
 
         ## Pop out iteratively the "words" of the path until it is an empty
         # list.
         ##  This way it should be easier to plug in more endpoint logic
+
         # Resource type
         resource_type = path.pop(0)
         if not path:
-            return (resource_type, page, pk, query_type)
```

Comment:
Done. Now you can only use uuid or starting pattern of uuid to refer to a specific node. N.B> load_node has been refactored